### PR TITLE
[Keyboard] Add ymd67 keyboard

### PR DESCRIPTION
**Pull request by SwiftLawnGnome**
## Description
Add support for the YMDK YMD67 keyboard.

Fun fact: since AliExpress says "1 sold", I'm currently the only person who owns this keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
> Fun fact: since AliExpress says "1 sold", I'm currently the only person who owns this keyboard.

:joy:

File: keyboards/ymd67/keymaps/default/readme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+![YMD67 layout image](https://imgur.com/a/mee1zG4)
```
Comment (noroadsleft) [inline keyboards/ymd67/keymaps/default/readme.md]: It's preferred to have the header first.

```suggestion
# Default YMD67 layout
```

File: keyboards/ymd67/keymaps/default/readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+# Default YMD67 layout
```
Comment (noroadsleft) [inline keyboards/ymd67/keymaps/default/readme.md]: This needs to be a direct image link, or the image doesn't show on the page when this file gets rendered.

```suggestion
![YMD67 layout image](https://i.imgur.com/YghD1LH.png)
```

File: keyboards/ymd67/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+A low-profile minila keyboard kit by [YMDK](https://ymdk.aliexpress.com/store/429151?spm=2114.10010108.0.0.3ab23641lIkgzm). Available in multiple colors and has RGB side-lighting.
+
+* Keyboard Maintainer: [Zach Shaftel](https://github.com/swiftlawngnome)
+* Hardware Supported: YMD67
+* Hardware Availability: [AliExpress](https://www.aliexpress.com/item/4000311811121.html)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ymd67:default
```
Comment (noroadsleft) [inline keyboards/ymd67/readme.md]:
```suggestion
    make ymdk/ymd67:default
```

File: keyboards/ymd67/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes # Virtual DIP switch configuration(+1000)
```
Comment (noroadsleft) [inline keyboards/ymd67/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = yes # Virtual DIP switch configuration
```

File: keyboards/ymd67/rules.mk [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+MOUSEKEY_ENABLE = no   # Mouse keys(+4700)
```
Comment (noroadsleft) [inline keyboards/ymd67/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no   # Mouse keys
```

File: keyboards/ymd67/rules.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+EXTRAKEY_ENABLE = yes  # Audio control and System control(+450)
```
Comment (noroadsleft) [inline keyboards/ymd67/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes  # Audio control and System control
```

File: keyboards/ymd67/rules.mk [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+CONSOLE_ENABLE = no    # Console for debug(+400)
```
Comment (noroadsleft) [inline keyboards/ymd67/rules.mk]:
```suggestion
CONSOLE_ENABLE = no    # Console for debug
```

File: keyboards/ymd67/rules.mk [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+COMMAND_ENABLE = yes   # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes      # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```
Comment (noroadsleft) [inline keyboards/ymd67/rules.mk]: There's a backlight pin defined, and the photos on the listing seem to show pads for in-switch LEDs.

```suggestion
BACKLIGHT_ENABLE = yes # Enable keyboard backlight functionality
```

File: keyboards/ymd67/ymd67.c [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#include "ymd67.h"
+
+void matrix_init_user(void) {}
+
+void matrix_scan_user(void) {}
```
Comment (noroadsleft) [inline keyboards/ymd67/ymd67.c]: Both of these functions should be deleted, since they aren't being used.

**Comment by SwiftLawnGnome** [comment]
Ok, changes applied, hope I didn't miss anything

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**